gdc_ng_models/models/batch.py

```diff
@@ -0,0 +1,131 @@
+import json
+import sqlalchemy
+from sqlalchemy import orm
+from sqlalchemy.ext import declarative
+
+Base = declarative.declarative_base()
+
+
+class Batch(Base):
+    """Batch class to represent a collection of nodes.
+
+    Attributes:
+        id: unique identifier for the batch
+        name: name given to the batch
+        description: optional description for the batch
+        project_id: project the batch is a part of
+        created_datetime: the date and time when the batch is created
+    """
+
+    __tablename__ = "batch"
+    __mapper_args__ = {"eager_defaults": True}
+
+    # start sequence at high number to avoid collisions with existing batch_id
+    id_seq = sqlalchemy.Sequence("batch_id_seq", metadata=Base.metadata, start=1000)
+
+    id = sqlalchemy.Column(
+        sqlalchemy.BigInteger,
+        nullable=False,
+        primary_key=True,
+        server_default=id_seq.next_value(),
+    )
+    name = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+    description = sqlalchemy.Column(sqlalchemy.Text, nullable=True)
+    project_id = sqlalchemy.Column(sqlalchemy.String(64), nullable=False, index=True)
```

Comment:
Where is the 64-character limit for project id coming from?

See above statement about indices.  For non-primary-key indices, the naming convention is `<table_name>_<column_names>_idx`. (Index names are limited to 63 characters.  So, some abbreviation would be required if we have a long index name)

Reply:
i was looking in the models directory at what other models were using for character limits, as i wasn't sure what the limit should be. For example, qcreport sets project_id to 64, but project_id sets it to 32. The inconsistencies exist for other columns as well (like node_id, which is Text in some models and String(64) in others).

Reply:
Will this be a problem if we expand batch across projects in the future?

Reply:
hm good point. project_id seems tied more to node, so we could make project_id another column in the batch membership table instead. so then batch could span multiple projects

Reply:
will keep project_id on Batch for now, and when we need to refactor down the line we can